htmd/protocols/production_v6.py

```diff
@@ -0,0 +1,422 @@
+                                   'Production.acemd.{}'.format(os.path.join(inputdir, userval), field))
+
+            if self.acemd.__dict__[field] is None:
+                for val in defaults[field]:
+                    if os.path.exists(os.path.join(inputdir, val)):
+                        self.acemd.__dict__[field] = val
+                        break
+
+            if userval is not None and self.acemd.__dict__[field] is not None and self.acemd.__dict__[field] != userval:
+                logger.warning('Could not locate structure file {}. Using {} instead.'.format(
+                    os.path.join(inputdir, userval), os.path.join(inputdir, self.acemd.__dict__[field])
+                ))
+            elif self.acemd.__dict__[field] is None:
+                raise RuntimeError('Could not locate any {f:} file in {i:}. '
+                                   'Please set the Production.acemd.{f:} property to '
+                                   'point to the {f:} file'.format(f=field, i=inputdir))
+
+    def _amberFixes(self):
+        # AMBER specific fixes
+        if self.acemd.structure.endswith('.prmtop'):
+            self.acemd.parmfile = self.acemd.parameters
+            self.acemd.parameters = None
+
+    def write(self, inputdir, outputdir):
+        """ Writes the production protocol and files into a folder.
+
+        Parameters
+        ----------
+        inputdir : str
+            Path to a directory containing the files produced by a equilibration process.
+        outputdir : str
+            Directory where to write the production setup files.
+        """
+        self._findFiles(inputdir)
+        self._amberFixes()
+
+        from htmd.units import convert
+        numsteps = convert(self.timeunits, 'timesteps', self.runtime, timestep=self.acemd.timestep)
+        self.acemd.temperature = self.temperature
+        self.acemd.langevintemp = self.temperature
+        self.acemd.thermostattemp = self.temperature
+        self.acemd.run = str(numsteps)
+        self.acemd.restraints = self.restraints
+        if self.adaptive:
+            self.acemd.binvelocities = None
+
+        self.acemd.setup(inputdir, outputdir, overwrite=True)
+
+
+if __name__ == "__main__":
+    from htmd.home import home
+    from htmd.util import tempname
+    import filecmp
+    from glob import glob
+
+    pdbid = '3PTB'
+    pd = Production()
+    pd.runtime = 4
+    pd.timeunits = 'ns'
+    pd.temperature = 350
+    pd.useconstraints = True
+    pd.constraints = {'protein and name CA': 1, 'protein and noh and not name CA': 0.1}
+    pd.fb_reference = 'protein and name CA'
+    pd.fb_selection = 'segname L and noh'
+    pd.fb_box = [-20, 20, -20, 20, 43, 45]
+    pd.fb_k = 5
+    tmpdir = tempname()
+    pd.write(home(dataDir=os.path.join('test-equilibration', pdbid, 'postrun')), tmpdir)
+
+    # Compare with reference
+    refdir = home(dataDir=os.path.join('test-production-v6', pdbid, 'prerun'))
```

Comment:
And 'postrun' here?

Reply:
I copied it from v5 which you must have written

Reply:
hum, let me check it

Reply:
No it makes sense. The one is equil post run and the other is production prerun

Reply:
yeah, you are right